File: app_pojav_zh/src/main/java/net/kdt/pojavlaunch/modloaders/modpacks/models/ModDetail.java
package net.kdt.pojavlaunch.modloaders.modpacks.models;

import com.movtery.pojavzh.ui.subassembly.downloadmod.ModVersionItem;

import java.util.List;

public class ModDetail extends ModItem {
    /* SHA 1 hashes, null if a hash is unavailable */
    public List<ModVersionItem> modVersionItems;

    public ModDetail(ModItem item, List<ModVersionItem> modVersionItems) {
        super(item.apiSource, item.isModpack, item.id, item.title, item.description, item.downloadCount, item.modloaders, item.imageUrl);
        this.modVersionItems = modVersionItems;
    }
}

File: app_pojav_zh/src/main/java/com/movtery/pojavzh/feature/mod/item/ModDetail.kt
package com.movtery.pojavzh.feature.mod.item

import com.movtery.pojavzh.ui.subassembly.downloadmod.ModVersionItem

class ModDetail(
    item: ModItem,
    /* SHA 1 hashes, null if a hash is unavailable */
    var modVersionItems: MutableList<ModVersionItem>?
) : ModItem(
    item.apiSource,
    item.isModpack,
    item.id,
    item.title,
    item.subTitle,
    item.description,
    item.downloadCount,
    item.categories,
    item.modloaders,
    item.imageUrl
)
